pta-intelligent-search-metadata-extractor/src/main/java/fi/maanmittauslaitos/pta/search/metadata/model/ResponsibleParty.java

```diff
@@ -1,28 +1,68 @@
 package fi.maanmittauslaitos.pta.search.metadata.model;
 
+import java.util.ArrayList;
 import java.util.HashMap;
+import java.util.List;
 import java.util.Map;
 
 public class ResponsibleParty {
-	private String organisationName;
 	private String isoRole;
-	private Map<String, String> localisedOrganisationName = new HashMap<>();
-	
-	
+	private List<String> email;
+	private String partyName;
+	private Map<String, String> localizedPartyName;
+
+	public ResponsibleParty() {
+		email = new ArrayList<>();
+		localizedPartyName = new HashMap<>();
+	}
+
+	public ResponsibleParty(ResponsibleParty responsibleParty) {
+		if (responsibleParty != null) {
+			this.email = responsibleParty.getEmail();
+			this.partyName = responsibleParty.getPartyName();
+			this.isoRole = responsibleParty.getIsoRole();
+			this.localizedPartyName = responsibleParty.getLocalizedPartyName();
+		}
+	}
+
+	/**
+	 * Deprecated in favor of more general semantics.
+	 *
+	 * @deprecated use {@link #setPartyName(String)} ()} instead.
+	 */
+	@Deprecated
 	public void setOrganisationName(String organisationName) {
```

Comment:
Add a general TODO to the class that we should get rid of the OrganisationName field as now there will be double fields in the Elasticsearch index. PartyName is better

Reply:
Added a TODO regarding this